### DISCUSSION TITLE
Caveat googlor - note to self about datasheets

### DISCUSSION THREAD

**Discussion by grandaspanna** [SpenceKonde/megaTinyCore#758] [Show and tell]
Was tearing my hair out (what little is left) when trying to reconcile an attiny412 behaviour with the datasheet, only to eventually realise that the first hit for "attiny 412 datasheet" in google returns the *preliminary* datasheet with some major issues, particularly in the portmux mapping table.

Just thought I'd share to save others similar grief.

**Comment by SpenceKonde** [comment]
Yeah, I've noticed that too

Unfortunately it's not the datasheet version where they screwed up and left in the OCD key as well. There's one preliminary datasheet somewhere that includes the key used to enable on-chip-debugging and I think one other neat mode that's otherwise only accessible through offciial tooling.

**Comment by ObviousInRetrospect** [comment]
Of course, Microchip confirmed to me recently in a support case that there is no non-preliminary data sheet for the DB series. And over there I know of at least one totally undocumented undisclosed limitation. So certainly check for a non-preliminary one but sometimes if the chip is only say a couple years old it might not exist.

**Comment by SpenceKonde** [comment]
Really? I swore there was - though  "data sheet clarifications" apply.

The big ones are:
1000 rewrite cycles on flash is all they guarantee (likely more under
non-adverse conditions)
EDGE in TCB selects between positive edge, and any edge, instead of
positive and negative like the datasheet says (the DA and DD have this
amended) This is of course an improvement since you can just invert the pin
if you want negative edges.
If EESAVE fuse is set, EEPROM is not erased on a locked chip when chip
erase happens
There;s some other datasheet clarifications about TCB, exotic power save
modes, and the DAC and output buffer. Note that most of those ARE in teh DA
datasheet, just not the DB one.

On Sat, Aug 6, 2022 at 10:28 AM ObviousInRetrospect <
***@***.***> wrote:

> Of course, Microchip confirmed to me recently in a support case that there
> is no non-preliminary data sheet for the DB series. And over there I know
> of at least one totally undocumented undisclosed limitation. So certainly
> check for a non-preliminary one but sometimes if the chip is only say a
> couple years old it might not exist.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/SpenceKonde/megaTinyCore/discussions/758#discussioncomment-3339730>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABTXEW5AWRGJFARSDB6RCHDVXZY7BANCNFSM55XYOPAA>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***
> com>
>


-- 
____________
Spence Konde
Azzy’S Electronics

New products! Check them out at tindie.com/stores/DrAzzy
GitHub: github.com/SpenceKonde
ATTinyCore <https://github.com/SpenceKonde/ATTinyCore>: Arduino support for
all pre-2016 tinyAVR with >2k flash!
megaTinyCore <https://github.com/SpenceKonde/megaTinyCore>: Arduino support
for all post-2016 tinyAVR parts!
DxCore <https://github.com/SpenceKonde/DxCore>: Arduino support for the AVR
Dx-series parts, the latest and greatest from Microchip!
Contact: ***@***.***

**Reply by ObviousInRetrospect** [reply]
"Only the datasheet that you are referring to is provided in the AVR64DB32 product page is available currently – 
https://ww1.microchip.com/downloads/aemDocuments/documents/MCU08/ProductDocuments/DataSheets/AVR64DB28-32-48-64-DataSheet-DS40002300A.pdf"
-Microchip, 7/19/2022

DS40002300A-page 618 says that Vadcref (ADC Reference Voltage) MIN is 1.8v. But page 611 says there is an internal 1.024v reference. 

The undocumented issues related to this are:
VREFA input pin voltage minimum is 1.8v for VDD < 2.7V but 1.024V for VDD >=2.7V
When using a VADCREF below 1.8 Fclk_adc must be <= 500kHz (internal or external)

This applies to DB and DD. I didn't ask about DA.

**Comment by SpenceKonde** [comment]
That's a beauty

On Sat, Aug 6, 2022 at 6:52 PM ObviousInRetrospect ***@***.***>
wrote:

> "Only the datasheet that you are referring to is provided in the AVR64DB32
> product page is available currently –
>
> https://ww1.microchip.com/downloads/aemDocuments/documents/MCU08/ProductDocuments/DataSheets/AVR64DB28-32-48-64-DataSheet-DS40002300A.pdf
> "
> -Microchip, 7/19/2022
>
> DS40002300A-page 618 says that Vadcref (ADC Reference Voltage) MIN is
> 1.8v. But page 611 says there is an internal 1.024v reference.
>
> The undocumented issues related to this are:
> VREFA input pin voltage minimum is 1.8v for VDD < 2.7V but 1.024V for VDD
> >=2.7V
> When using a VADCREF below 1.8 Fclk_adc must be <= 500kHz (internal or
> external)
>
> This applies to DB and DD. I didn't ask about DA.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/SpenceKonde/megaTinyCore/discussions/758#discussioncomment-3341228>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABTXEW5J6JAHQDZM6IDXJATVX3UCJANCNFSM55XYOPAA>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***
> com>
>


-- 
____________
Spence Konde
Azzy’S Electronics

New products! Check them out at tindie.com/stores/DrAzzy
GitHub: github.com/SpenceKonde
ATTinyCore <https://github.com/SpenceKonde/ATTinyCore>: Arduino support for
all pre-2016 tinyAVR with >2k flash!
megaTinyCore <https://github.com/SpenceKonde/megaTinyCore>: Arduino support
for all post-2016 tinyAVR parts!
DxCore <https://github.com/SpenceKonde/DxCore>: Arduino support for the AVR
Dx-series parts, the latest and greatest from Microchip!
Contact: ***@***.***
